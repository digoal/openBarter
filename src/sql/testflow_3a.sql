-- SET search_path TO test;
truncate torder;
truncate tstack;
truncate tmvt;
truncate towner;
SELECT setval('tstack_id_seq',1,false);

copy torder from '/home/olivier/ob92/src/sql/torder_test_10000.sql';
copy towner from '/home/olivier/ob92/src/sql/towner_test_10000.sql';
truncate tstack;
SELECT setval('tstack_id_seq',10000,true);

select * from fsubmitquote(1,'own82','qlt22','qlt23');select * from fproducemvt();
select * from fsubmitbarter(1,'own82',NULL,'qlt22',67432,'qlt23',30183,30183);select * from fproducemvt();

select * from fsubmitquote(2,'own82','qlt22','qlt23');select * from fproducemvt(); 
select * from fsubmitbarter(2,'own82',NULL,'qlt22',61017,'qlt23',45276,45276);select * from fproducemvt();
select id,type,json from tmvt where type >3;




