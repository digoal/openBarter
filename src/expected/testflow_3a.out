SET search_path TO market;
truncate torder;
truncate tstack;
truncate tmvt;
truncate towner;
SELECT setval('tstack_id_seq',1,false);
 setval 
--------
      1
(1 row)

copy torder(usr,ord,created,updated) from '/home/olivier/ob92/src/sql/torder_test_10000.sql';
copy towner from '/home/olivier/ob92/src/sql/towner_test_10000.sql';
truncate tstack;
SELECT setval('tstack_id_seq',10000,true);
 setval 
--------
  10000
(1 row)

select * from fsubmitquote(1,'own82','qlt22','qlt23');select * from fproducemvt();
  id   | diag 
-------+------
 10001 |    0
(1 row)

                  ord                   |    ordp    | qtt_requ | qtt_prov |  qtt  | qtt_reci | qtt_give | err |                                                       json                                                       
----------------------------------------+------------+----------+----------+-------+----------+----------+-----+------------------------------------------------------------------------------------------------------------------
 (141,10001,72,10001,1,qlt22,1,qlt23,1) | (,,,,,,,,) |    67432 |    30183 | 30182 |    67430 |    30182 |   0 | {"qtt_requ":67432,"qtt_prov":30183,"qtt":30182,"qtt_reci":67430,"qtt_give":30182,"paths":[                      +
                                        |            |          |          |       |          |          |     | [{"type":2, "id":8928, "oid":8928, "own":72, "qtt_requ":49263, "qtt_prov":87732, "qtt":87732, "flowr":53750},   +
                                        |            |          |          |       |          |          |     | {"type":1, "id":515, "oid":515, "own":69, "qtt_requ":53751, "qtt_prov":67432, "qtt":67432, "flowr":67430},      +
                                        |            |          |          |       |          |          |     | {"type":141, "id":10001, "oid":10001, "own":72, "qtt_requ":67432, "qtt_prov":30183, "qtt":30182, "flowr":30182}]+
                                        |            |          |          |       |          |          |     | ]}
(1 row)

select * from fsubmitbarter(1,'own82',NULL,'qlt22',67432,'qlt23',30183,30182,'1 hour'::interval);select * from fproducemvt();
  id   | diag 
-------+------
 10002 |    0
(1 row)

                       ord                        |    ordp    | qtt_requ | qtt_prov |  qtt  | qtt_reci | qtt_give | err | json 
--------------------------------------------------+------------+----------+----------+-------+----------+----------+-----+------
 (1,10002,72,10002,67432,qlt22,30183,qlt23,30182) | (,,,,,,,,) |    67432 |    30183 | 30182 |    67430 |    30182 |   0 | 
(1 row)

select xid,own_src,own_dst,qtt,nat from tmvt order by id desc limit 3;
  xid  | own_src | own_dst |  qtt  |  nat  
-------+---------+---------+-------+-------
   515 | own91   | own82   | 67430 | qlt22
  8928 | own82   | own91   | 53750 | qlt53
 10002 | own82   | own82   | 30182 | qlt23
(3 rows)

select * from fsubmitquote(1,'own82','qlt22','qlt10');select * from fproducemvt();
  id   | diag 
-------+------
 10003 |    0
(1 row)

                  ord                   |    ordp    | qtt_requ | qtt_prov | qtt  | qtt_reci | qtt_give | err |                                                     json                                                      
----------------------------------------+------------+----------+----------+------+----------+----------+-----+---------------------------------------------------------------------------------------------------------------
 (141,10003,72,10003,1,qlt22,1,qlt10,1) | (,,,,,,,,) |    49252 |     2177 | 2176 |    49237 |     2176 |   0 | {"qtt_requ":49252,"qtt_prov":2177,"qtt":2176,"qtt_reci":49237,"qtt_give":2176,"paths":[                      +
                                        |            |          |          |      |          |          |     | [{"type":2, "id":7260, "oid":7260, "own":95, "qtt_requ":2176, "qtt_prov":78849, "qtt":78849, "flowr":78837}, +
                                        |            |          |          |      |          |          |     | {"type":2, "id":754, "oid":754, "own":61, "qtt_requ":81762, "qtt_prov":51072, "qtt":51072, "flowr":49237},   +
                                        |            |          |          |      |          |          |     | {"type":141, "id":10003, "oid":10003, "own":72, "qtt_requ":49252, "qtt_prov":2177, "qtt":2176, "flowr":2176}]+
                                        |            |          |          |      |          |          |     | ]}
(1 row)

select * from fsubmitquote(1,'own82','qlt22',49252,'qlt10',2177,2176);select * from fproducemvt();
  id   | diag 
-------+------
 10004 |    0
(1 row)

                       ord                        |    ordp    | qtt_requ | qtt_prov | qtt  | qtt_reci | qtt_give | err |                                                     json                                                      
--------------------------------------------------+------------+----------+----------+------+----------+----------+-----+---------------------------------------------------------------------------------------------------------------
 (129,10004,72,10004,49252,qlt22,2177,qlt10,2176) | (,,,,,,,,) |    49252 |     2177 | 2176 |    49237 |     2176 |   0 | {"qtt_requ":49252,"qtt_prov":2177,"qtt":2176,"qtt_reci":49237,"qtt_give":2176,"paths":[                      +
                                                  |            |          |          |      |          |          |     | [{"type":2, "id":7260, "oid":7260, "own":95, "qtt_requ":2176, "qtt_prov":78849, "qtt":78849, "flowr":78837}, +
                                                  |            |          |          |      |          |          |     | {"type":2, "id":754, "oid":754, "own":61, "qtt_requ":81762, "qtt_prov":51072, "qtt":51072, "flowr":49237},   +
                                                  |            |          |          |      |          |          |     | {"type":129, "id":10004, "oid":10004, "own":72, "qtt_requ":49252, "qtt_prov":2177, "qtt":2176, "flowr":2176}]+
                                                  |            |          |          |      |          |          |     | ]}
(1 row)

select * from fsubmitbarter(1,'own82',NULL,'qlt22',49252,'qlt10',2177,2176,'1 hour'::interval);select * from fproducemvt(); 
  id   | diag 
-------+------
 10005 |    0
(1 row)

                      ord                       |    ordp    | qtt_requ | qtt_prov | qtt  | qtt_reci | qtt_give | err | json 
------------------------------------------------+------------+----------+----------+------+----------+----------+-----+------
 (1,10005,72,10005,49252,qlt22,2177,qlt10,2176) | (,,,,,,,,) |    49252 |     2177 | 2176 |    49237 |     2176 |   0 | 
(1 row)

select xid,own_src,own_dst,qtt,nat from tmvt order by id desc limit 3;
  xid  | own_src | own_dst |  qtt  |  nat  
-------+---------+---------+-------+-------
   754 | own7    | own82   | 49237 | qlt22
  7260 | own77   | own7    | 78837 | qlt95
 10005 | own82   | own77   |  2176 | qlt10
(3 rows)

select * from fsubmitquote(1,'own82','qlt2',60000,'qlt23',45276,45276);select * from fproducemvt();
  id   | diag 
-------+------
 10006 |    0
(1 row)

                        ord                        |    ordp    | qtt_requ | qtt_prov |  qtt  | qtt_reci | qtt_give | err |                                                       json                                                        
---------------------------------------------------+------------+----------+----------+-------+----------+----------+-----+-------------------------------------------------------------------------------------------------------------------
 (129,10006,72,10006,60000,qlt2,45276,qlt23,45276) | (,,,,,,,,) |    60000 |    45276 | 45276 |    82755 |    45276 |   0 | {"qtt_requ":60000,"qtt_prov":45276,"qtt":45276,"qtt_reci":82755,"qtt_give":45276,"paths":[                       +
                                                   |            |          |          |       |          |          |     | [{"type":2, "id":4352, "oid":4352, "own":36, "qtt_requ":2313, "qtt_prov":43484, "qtt":43484, "flowr":43484},     +
                                                   |            |          |          |       |          |          |     | {"type":1, "id":5, "oid":5, "own":5, "qtt_requ":45499, "qtt_prov":45464, "qtt":45464, "flowr":17954},            +
                                                   |            |          |          |       |          |          |     | {"type":129, "id":10006, "oid":10006, "own":72, "qtt_requ":60000, "qtt_prov":45276, "qtt":45276, "flowr":5598}], +
                                                   |            |          |          |       |          |          |     | [{"type":2, "id":1046, "oid":1046, "own":78, "qtt_requ":2318, "qtt_prov":11725, "qtt":11725, "flowr":11725},     +
                                                   |            |          |          |       |          |          |     | {"type":1, "id":345, "oid":345, "own":79, "qtt_requ":43486, "qtt_prov":31371, "qtt":31371, "flowr":6035},        +
                                                   |            |          |          |       |          |          |     | {"type":129, "id":10006, "oid":10006, "own":72, "qtt_requ":60000, "qtt_prov":45276, "qtt":39678, "flowr":3249}], +
                                                   |            |          |          |       |          |          |     | [{"type":1, "id":2361, "oid":2361, "own":2, "qtt_requ":59716, "qtt_prov":98494, "qtt":98494, "flowr":30997},     +
                                                   |            |          |          |       |          |          |     | {"type":1, "id":337, "oid":337, "own":44, "qtt_requ":25168, "qtt_prov":37776, "qtt":37776, "flowr":37776},       +
                                                   |            |          |          |       |          |          |     | {"type":129, "id":10006, "oid":10006, "own":72, "qtt_requ":60000, "qtt_prov":45276, "qtt":36429, "flowr":23145}],+
                                                   |            |          |          |       |          |          |     | [{"type":1, "id":2361, "oid":2361, "own":2, "qtt_requ":59716, "qtt_prov":98494, "qtt":67497, "flowr":18376},     +
                                                   |            |          |          |       |          |          |     | {"type":1, "id":321, "oid":321, "own":48, "qtt_requ":53964, "qtt_prov":73494, "qtt":73494, "flowr":20990},       +
                                                   |            |          |          |       |          |          |     | {"type":129, "id":10006, "oid":10006, "own":72, "qtt_requ":60000, "qtt_prov":45276, "qtt":13284, "flowr":13284}] +
                                                   |            |          |          |       |          |          |     | ]}
(1 row)

select * from fsubmitbarter(1,'own82',NULL,'qlt2',60000,'qlt23',45276,45276,'1 hour'::interval);select * from fproducemvt();
  id   | diag 
-------+------
 10007 |    0
(1 row)

                       ord                       |    ordp    | qtt_requ | qtt_prov |  qtt  | qtt_reci | qtt_give | err | json 
-------------------------------------------------+------------+----------+----------+-------+----------+----------+-----+------
 (1,10007,72,10007,60000,qlt2,45276,qlt23,45276) | (,,,,,,,,) |    60000 |    45276 | 45276 |    82755 |    45276 |   0 | 
(1 row)

select id,nbt,nbc,xid,own_src,own_dst,qtt,nat from tmvt order by id desc limit 12;
 id | nbt | nbc |  xid  | own_src | own_dst |  qtt  |  nat  
----+-----+-----+-------+---------+---------+-------+-------
 30 |  12 |   3 |   321 | own35   | own82   | 20990 | qlt2
 29 |  12 |   3 |  2361 | own66   | own35   | 18376 | qlt72
 28 |  12 |   3 | 10007 | own82   | own66   | 13284 | qlt23
 27 |  12 |   3 |   337 | own37   | own82   | 37776 | qlt2
 26 |  12 |   3 |  2361 | own66   | own37   | 30997 | qlt72
 25 |  12 |   3 | 10007 | own82   | own66   | 23145 | qlt23
 24 |  12 |   3 |   345 | own56   | own82   |  6035 | qlt2
 23 |  12 |   3 |  1046 | own43   | own56   | 11725 | qlt92
 22 |  12 |   3 | 10007 | own82   | own43   |  3249 | qlt23
 21 |  12 |   3 |     5 | own3    | own82   | 17954 | qlt2
 20 |  12 |   3 |  4352 | own75   | own3    | 43484 | qlt89
 19 |  12 |   3 | 10007 | own82   | own75   |  5598 | qlt23
(12 rows)

select sum(qtt) from tmvt where own_src='own82' and nat='qlt23' and id>18;
  sum  
-------
 45276
(1 row)

select sum(qtt) from tmvt where own_dst='own82' and nat='qlt2' and id>18;
  sum  
-------
 82755
(1 row)

select * from fsubmitquote(1,'own1','qlt1','qlt2');select * from fproducemvt();
  id   | diag 
-------+------
 10008 |    0
(1 row)

                 ord                  |    ordp    | qtt_requ | qtt_prov | qtt  | qtt_reci | qtt_give | err |                                                     json                                                      
--------------------------------------+------------+----------+----------+------+----------+----------+-----+---------------------------------------------------------------------------------------------------------------
 (141,10008,93,10008,1,qlt1,1,qlt2,1) | (,,,,,,,,) |    30822 |     9667 | 9666 |    30820 |     9666 |   0 | {"qtt_requ":30822,"qtt_prov":9667,"qtt":9666,"qtt_reci":30820,"qtt_give":9666,"paths":[                      +
                                      |            |          |          |      |          |          |     | [{"type":2, "id":6026, "oid":6026, "own":78, "qtt_requ":84630, "qtt_prov":83977, "qtt":83977, "flowr":9592}, +
                                      |            |          |          |      |          |          |     | {"type":1, "id":85, "oid":85, "own":29, "qtt_requ":9592, "qtt_prov":30822, "qtt":30822, "flowr":30820},      +
                                      |            |          |          |      |          |          |     | {"type":141, "id":10008, "oid":10008, "own":93, "qtt_requ":30822, "qtt_prov":9667, "qtt":9666, "flowr":9666}]+
                                      |            |          |          |      |          |          |     | ]}
(1 row)

select * from fsubmitquote(1,'own1','qlt1',30822,'qlt2',9667);select * from fproducemvt();
  id   | diag 
-------+------
 10009 |    0
(1 row)

                     ord                     |    ordp    | qtt_requ | qtt_prov | qtt  | qtt_reci | qtt_give | err |                                                     json                                                      
---------------------------------------------+------------+----------+----------+------+----------+----------+-----+---------------------------------------------------------------------------------------------------------------
 (133,10009,93,10009,30822,qlt1,9667,qlt2,0) | (,,,,,,,,) |    30822 |     9667 | 7563 |    30819 |     7563 |   0 | {"qtt_requ":30822,"qtt_prov":9667,"qtt":7563,"qtt_reci":30819,"qtt_give":7563,"paths":[                      +
                                             |            |          |          |      |          |          |     | [{"type":1, "id":5071, "oid":5071, "own":13, "qtt_requ":10428, "qtt_prov":21601, "qtt":21601, "flowr":12258},+
                                             |            |          |          |      |          |          |     | {"type":1, "id":85, "oid":85, "own":29, "qtt_requ":9592, "qtt_prov":30822, "qtt":30822, "flowr":30819},      +
                                             |            |          |          |      |          |          |     | {"type":133, "id":10009, "oid":10009, "own":93, "qtt_requ":30822, "qtt_prov":9667, "qtt":7563, "flowr":7563}]+
                                             |            |          |          |      |          |          |     | ]}
(1 row)

select * from fsubmitquote(1,'own1','qlt1',30822,'qlt2',9667,7563);select * from fproducemvt();
  id   | diag 
-------+------
 10010 |    0
(1 row)

                      ord                       |    ordp    | qtt_requ | qtt_prov | qtt  | qtt_reci | qtt_give | err |                                                     json                                                      
------------------------------------------------+------------+----------+----------+------+----------+----------+-----+---------------------------------------------------------------------------------------------------------------
 (129,10010,93,10010,30822,qlt1,9667,qlt2,7563) | (,,,,,,,,) |    30822 |     9667 | 7563 |    30819 |     7563 |   0 | {"qtt_requ":30822,"qtt_prov":9667,"qtt":7563,"qtt_reci":30819,"qtt_give":7563,"paths":[                      +
                                                |            |          |          |      |          |          |     | [{"type":1, "id":5071, "oid":5071, "own":13, "qtt_requ":10428, "qtt_prov":21601, "qtt":21601, "flowr":12258},+
                                                |            |          |          |      |          |          |     | {"type":1, "id":85, "oid":85, "own":29, "qtt_requ":9592, "qtt_prov":30822, "qtt":30822, "flowr":30819},      +
                                                |            |          |          |      |          |          |     | {"type":129, "id":10010, "oid":10010, "own":93, "qtt_requ":30822, "qtt_prov":9667, "qtt":7563, "flowr":7563}]+
                                                |            |          |          |      |          |          |     | ]}
(1 row)

select * from fsubmitbarter(1,'own1',NULL,'qlt1',30822,'qlt2',9667,7563,'1 hour'::interval);select * from fproducemvt();
  id   | diag 
-------+------
 10011 |    0
(1 row)

                     ord                      |    ordp    | qtt_requ | qtt_prov | qtt  | qtt_reci | qtt_give | err | json 
----------------------------------------------+------------+----------+----------+------+----------+----------+-----+------
 (1,10011,93,10011,30822,qlt1,9667,qlt2,7563) | (,,,,,,,,) |    30822 |     9667 | 7563 |    30819 |     7563 |   0 | 
(1 row)

select id,nbt,nbc,xid,own_src,own_dst,qtt,nat from tmvt d where id>33;
 id | nbt | nbc |  xid  | own_src | own_dst |  qtt  |  nat  
----+-----+-----+-------+---------+---------+-------+-------
 34 |   3 |   3 | 10011 | own1    | own15   |  7563 | qlt2
 35 |   3 |   3 |  5071 | own15   | own58   | 12258 | qlt95
 36 |   3 |   3 |    85 | own58   | own1    | 30819 | qlt1
(3 rows)

select sum(qtt) from tmvt where own_src='own1' and nat='qlt2' and id>33;
 sum  
------
 7563
(1 row)

select sum(qtt) from tmvt where own_dst='own1' and nat='qlt1' and id>33;
  sum  
-------
 30819
(1 row)

-- 
select * from fsubmitquote(1,'own82','qlt25','qlt10');select * from fproducemvt();
  id   | diag 
-------+------
 10012 |    0
(1 row)

                  ord                   |    ordp    | qtt_requ | qtt_prov | qtt  | qtt_reci | qtt_give | err |                                                     json                                                      
----------------------------------------+------------+----------+----------+------+----------+----------+-----+---------------------------------------------------------------------------------------------------------------
 (141,10012,72,10012,1,qlt25,1,qlt10,1) | (,,,,,,,,) |    82800 |     3020 | 3019 |    82778 |     3019 |   0 | {"qtt_requ":82800,"qtt_prov":3020,"qtt":3019,"qtt_reci":82778,"qtt_give":3019,"paths":[                      +
                                        |            |          |          |      |          |          |     | [{"type":2, "id":9178, "oid":9178, "own":48, "qtt_requ":13928, "qtt_prov":62970, "qtt":62970, "flowr":13648},+
                                        |            |          |          |      |          |          |     | {"type":1, "id":245, "oid":245, "own":27, "qtt_requ":13651, "qtt_prov":82800, "qtt":82800, "flowr":82778},   +
                                        |            |          |          |      |          |          |     | {"type":141, "id":10012, "oid":10012, "own":72, "qtt_requ":82800, "qtt_prov":3020, "qtt":3019, "flowr":3019}]+
                                        |            |          |          |      |          |          |     | ]}
(1 row)

select * from fsubmitprequote('own82','qlt25','qlt10');select * from fproducemvt();
  id   | diag 
-------+------
 10013 |    0
(1 row)

                  ord                  |    ordp    | qtt_requ | qtt_prov | qtt | qtt_reci | qtt_give | err |                                                       json                                                       
---------------------------------------+------------+----------+----------+-----+----------+----------+-----+------------------------------------------------------------------------------------------------------------------
 (78,10013,72,10013,1,qlt25,1,qlt10,1) | (,,,,,,,,) |        0 |        0 |   0 |   387304 |   320195 |   0 | {"qtt_requ":0,"qtt_prov":0,"qtt":0,"qtt_reci":387304,"qtt_give":320195,"paths":[                                +
                                       |            |          |          |     |          |          |     | [{"type":2, "id":9178, "oid":9178, "own":48, "qtt_requ":13928, "qtt_prov":62970, "qtt":62970, "flowr":13648},   +
                                       |            |          |          |     |          |          |     | {"type":1, "id":245, "oid":245, "own":27, "qtt_requ":13651, "qtt_prov":82800, "qtt":82800, "flowr":82778},      +
                                       |            |          |          |     |          |          |     | {"type":78, "id":10013, "oid":10013, "own":72, "qtt_requ":82800, "qtt_prov":3020, "qtt":3019, "flowr":3019}],   +
                                       |            |          |          |     |          |          |     | [{"type":2, "id":6378, "oid":6378, "own":4, "qtt_requ":5391, "qtt_prov":80581, "qtt":80581, "flowr":80576},     +
                                       |            |          |          |     |          |          |     | {"type":1, "id":275, "oid":275, "own":92, "qtt_requ":94509, "qtt_prov":92831, "qtt":92831, "flowr":79140},      +
                                       |            |          |          |     |          |          |     | {"type":78, "id":10013, "oid":10013, "own":72, "qtt_requ":79150, "qtt_prov":5392, "qtt":5391, "flowr":5391}],   +
                                       |            |          |          |     |          |          |     | [{"type":2, "id":6618, "oid":6618, "own":62, "qtt_requ":59916, "qtt_prov":29686, "qtt":29686, "flowr":4},       +
                                       |            |          |          |     |          |          |     | {"type":1, "id":245, "oid":245, "own":27, "qtt_requ":13651, "qtt_prov":82800, "qtt":22, "flowr":20},            +
                                       |            |          |          |     |          |          |     | {"type":78, "id":10013, "oid":10013, "own":72, "qtt_requ":22, "qtt_prov":8, "qtt":7, "flowr":7}],               +
                                       |            |          |          |     |          |          |     | [{"type":1, "id":8985, "oid":8985, "own":21, "qtt_requ":24512, "qtt_prov":64193, "qtt":64193, "flowr":64192},   +
                                       |            |          |          |     |          |          |     | {"type":2, "id":1082, "oid":1082, "own":77, "qtt_requ":86141, "qtt_prov":93576, "qtt":93576, "flowr":69731},    +
                                       |            |          |          |     |          |          |     | {"type":78, "id":10013, "oid":10013, "own":72, "qtt_requ":69733, "qtt_prov":24513, "qtt":24512, "flowr":24512}],+
                                       |            |          |          |     |          |          |     | [{"type":2, "id":9944, "oid":9944, "own":52, "qtt_requ":14237, "qtt_prov":66031, "qtt":66031, "flowr":66029},   +
                                       |            |          |          |     |          |          |     | {"type":2, "id":814, "oid":814, "own":62, "qtt_requ":94119, "qtt_prov":55329, "qtt":55329, "flowr":38815},      +
                                       |            |          |          |     |          |          |     | {"type":78, "id":10013, "oid":10013, "own":72, "qtt_requ":38817, "qtt_prov":14238, "qtt":14237, "flowr":14237}],+
                                       |            |          |          |     |          |          |     | [{"type":1, "id":9053, "oid":9053, "own":17, "qtt_requ":31651, "qtt_prov":78394, "qtt":78394, "flowr":28089},   +
                                       |            |          |          |     |          |          |     | {"type":2, "id":814, "oid":814, "own":62, "qtt_requ":94119, "qtt_prov":55329, "qtt":16514, "flowr":16512},      +
                                       |            |          |          |     |          |          |     | {"type":78, "id":10013, "oid":10013, "own":72, "qtt_requ":16514, "qtt_prov":11342, "qtt":11341, "flowr":11341}],+
                                       |            |          |          |     |          |          |     | [{"type":1, "id":9053, "oid":9053, "own":17, "qtt_requ":31651, "qtt_prov":78394, "qtt":50305, "flowr":2},       +
                                       |            |          |          |     |          |          |     | {"type":2, "id":814, "oid":814, "own":62, "qtt_requ":94119, "qtt_prov":55329, "qtt":2, "flowr":1},              +
                                       |            |          |          |     |          |          |     | {"type":78, "id":10013, "oid":10013, "own":72, "qtt_requ":2, "qtt_prov":2, "qtt":1, "flowr":1}],                +
                                       |            |          |          |     |          |          |     | [{"type":2, "id":328, "oid":328, "own":26, "qtt_requ":60608, "qtt_prov":68128, "qtt":68128, "flowr":21950},     +
                                       |            |          |          |     |          |          |     | {"type":2, "id":1082, "oid":1082, "own":77, "qtt_requ":86141, "qtt_prov":93576, "qtt":23845, "flowr":23844},    +
                                       |            |          |          |     |          |          |     | {"type":78, "id":10013, "oid":10013, "own":72, "qtt_requ":23845, "qtt_prov":19528, "qtt":19527, "flowr":19527}],+
                                       |            |          |          |     |          |          |     | [{"type":2, "id":3298, "oid":3298, "own":25, "qtt_requ":87502, "qtt_prov":61824, "qtt":61824, "flowr":41374},   +
                                       |            |          |          |     |          |          |     | {"type":2, "id":1090, "oid":1090, "own":91, "qtt_requ":41375, "qtt_prov":55121, "qtt":55121, "flowr":55120},    +
                                       |            |          |          |     |          |          |     | {"type":78, "id":10013, "oid":10013, "own":72, "qtt_requ":55121, "qtt_prov":58560, "qtt":58559, "flowr":58559}],+
                                       |            |          |          |     |          |          |     | [{"type":1, "id":869, "oid":869, "own":70, "qtt_requ":21129, "qtt_prov":12354, "qtt":12354, "flowr":12354},     +
                                       |            |          |          |     |          |          |     | {"type":78, "id":10013, "oid":10013, "own":72, "qtt_requ":12354, "qtt_prov":21130, "qtt":21129, "flowr":21129}],+
                                       |            |          |          |     |          |          |     | [{"type":2, "id":300, "oid":300, "own":45, "qtt_requ":20995, "qtt_prov":33934, "qtt":33934, "flowr":33932},     +
                                       |            |          |          |     |          |          |     | {"type":1, "id":863, "oid":863, "own":68, "qtt_requ":98782, "qtt_prov":11450, "qtt":11450, "flowr":3933},       +
                                       |            |          |          |     |          |          |     | {"type":78, "id":10013, "oid":10013, "own":72, "qtt_requ":3933, "qtt_prov":20996, "qtt":20995, "flowr":20995}], +
                                       |            |          |          |     |          |          |     | [{"type":1, "id":3207, "oid":3207, "own":5, "qtt_requ":50038, "qtt_prov":31658, "qtt":31658, "flowr":31656},    +
                                       |            |          |          |     |          |          |     | {"type":1, "id":863, "oid":863, "own":68, "qtt_requ":98782, "qtt_prov":11450, "qtt":7517, "flowr":3669},        +
                                       |            |          |          |     |          |          |     | {"type":78, "id":10013, "oid":10013, "own":72, "qtt_requ":3669, "qtt_prov":50038, "qtt":50037, "flowr":50037}], +
                                       |            |          |          |     |          |          |     | [{"type":1, "id":1141, "oid":1141, "own":28, "qtt_requ":91440, "qtt_prov":11970, "qtt":11970, "flowr":11969},   +
                                       |            |          |          |     |          |          |     | {"type":1, "id":863, "oid":863, "own":68, "qtt_requ":98782, "qtt_prov":11450, "qtt":3848, "flowr":1387},        +
                                       |            |          |          |     |          |          |     | {"type":78, "id":10013, "oid":10013, "own":72, "qtt_requ":1387, "qtt_prov":91441, "qtt":91440, "flowr":91440}]  +
                                       |            |          |          |     |          |          |     | ]}
(1 row)

